### PR TITLE
Add sorting options to the program question bank.

### LINKED ISSUE #944: Let CiviForm admins sort the list of questions in the question bank by different factors
This can help surface the most popular questions more quickly.

This functionality should appear on both the Question bank page and in the form builder interface (in the sidebar there).

### PR DESCRIPTION
**Pull request by reemaka**
### Description

Adds the ability to sort questions in the question bank shown when adding questions to a program. Extends #5201, which added sorting options to the Questions page. As part of this, does some refactoring.

- Renames QuestionBank.java to ProgramQuestionBank.java to  better convey that the class handles rendering the question bank in the program edit view specifically.
- Creates a new class QuestionBank.java to hold elements shared between ProgramQuestionBank and QuestionsListView, including the sorting dropdown.
- Changes the styling of the search bar to be consistent between both views.

#### General

Read the full guidelines for PRs [here](https://docs.civiform.us/contributor-guide/developer-guide/technical-contribution-guide#guidelines-for-pull-requests)

- [x] Added the correct label: < feature | bug | dependencies | infrastructure | ignore-for-release | database >
- [x] Created unit and/or browser tests which fail without the change (if possible)
- [x] Performed manual testing (Chrome and Firefox if it includes front-end changes)

### Instructions for manual testing

Go to the edit page for a program and open the question bank by clicking "Add a question". Find the "Sort by" dropdown and check that the sorting is working as expected.

### Issue(s) this completes

Fixes #944.

### CODE REVIEW

File: server/app/views/components/ProgramQuestionBank.java
Comment (reemaka) [inline server/app/views/components/ProgramQuestionBank.java:1]: Diff: https://www.diffchecker.com/BIoVWVzS/

**Comment by michaelzetune** [comment]
Will look first thing in the morning!

File: server/app/views/components/QuestionBank.java [hunk 1]
```diff
@@ -1,135 +1,30 @@
...
-                        "w-6",
-                        "h-6",
-                        "cursor-pointer",
-                        "mr-2",
-                        ReferenceClasses.CLOSE_QUESTION_BANK_BUTTON))
-            .with(headerDiv));
-
-    LabelTag searchLabel =
-        label().withText("Search questions").withClasses("mx-2", "text-gray-500");
+  public static final DivTag renderFilter() {
```
Comment (michaelzetune) [inline server/app/views/components/QuestionBank.java:21]: Does this still need to be public?

File: server/app/views/components/QuestionSortOption.java [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+  // Display string for the select tag option for the option that sorts by this attribute in
+  // ascending order. Not present if this attribute should not have an ascending option.
+  private final Optional<String> displayStringAscending;
+
+  private QuestionSortOption(
+      String dataAttribute,
+      Optional<String> displayStringAscending,
+      Optional<String> displayStringDescending) {
+    this.dataAttribute = dataAttribute;
+    Preconditions.checkArgument(
```
Comment (michaelzetune) [inline server/app/views/components/QuestionSortOption.java:36]: Nit: static import `checkArgument`

File: server/app/views/components/QuestionSortOption.java [hunk 3]
```diff
@@ -0,0 +1,65 @@
+package views.components;
+
+import com.google.common.base.Preconditions;
+import com.google.common.collect.ImmutableList;
+import java.util.Optional;
+
+/** Represents a sorting option for a list of Questions. */
+public enum QuestionSortOption {
+  LAST_MODIFIED("lastmodified", Optional.empty(), Optional.of("Last modified")),
```
Comment (michaelzetune) [inline server/app/views/components/QuestionSortOption.java:9]: Can we have inline comments i.e. `/* dataAttribute */` to make clear what each param is?

File: browser-test/image_snapshots/admin_program_creation_test/question-bank-empty.png
Comment (michaelzetune) [inline browser-test/image_snapshots/admin_program_creation_test/question-bank-empty.png:1]: Thoughts on making the 'Not finding...' string left centered?

Also I think the "Create New Question" button would be better as a hyperlink than button, but I realize that's out of scope here.

**Review by michaelzetune — CHANGES_REQUESTED** [review]

File: server/app/views/components/QuestionSortOption.java (hunk 3 above)
Comment (reemaka) [reply server/app/views/components/QuestionSortOption.java:9]: Done

File: server/app/views/components/QuestionSortOption.java (hunk 2 above)
Comment (reemaka) [reply server/app/views/components/QuestionSortOption.java:36]: Done

File: server/app/views/components/ProgramQuestionBank.java [hunk 4]
```diff
@@ -0,0 +1,293 @@
...
+
+  // Url parameter used to force question bank open upon initial rendering
+  // of program edit page.
+  private static final String SHOW_QUESTION_BANK_PARAM = "sqb";
+
+  private final ProgramQuestionBankParams params;
+
+  /**
+   * Possible states of question bank upon rendering. Normally it starts hidden and triggered by
+   * user clicking "Add question" button. BUt in some cases, like returning to the program edit page
```
[the remark this answers is not in the document]
Comment (reemaka) [reply server/app/views/components/ProgramQuestionBank.java:44]: Done

File: browser-test/image_snapshots/admin_program_creation_test/question-bank-empty.png
Comment (reemaka) [reply browser-test/image_snapshots/admin_program_creation_test/question-bank-empty.png:1]: Discussed in Slack, leaving as is for now.

**Review by michaelzetune — APPROVED** [review]

**Pull request merged**